test/integration/fake_upstream.h

```diff
@@ -380,14 +372,7 @@ class FakeConnectionBase : public Logger::Loggable<Logger::Id::testing> {
   testing::AssertionResult
   waitForHalfClose(std::chrono::milliseconds timeout = TestUtility::DefaultTimeout);
 
-  ABSL_MUST_USE_RESULT
-  virtual testing::AssertionResult initialize() {
-    initialized_ = true;
-    return testing::AssertionSuccess();
-  }
-  ABSL_MUST_USE_RESULT
-  testing::AssertionResult
-  enableHalfClose(bool enabled, std::chrono::milliseconds timeout = TestUtility::DefaultTimeout);
+  virtual void initialize() { initialized_ = true; }
```

Comment:
Should `initialized_` have a lock annotation and/or should this function have some thread safety assert?

Reply:
Added thread annotations.